test_scripts/Polices/External_UCS/External_Consent_OFF/024_ATF_Policies_External_Consent_OFF_many_entities_pairs_in_one_group_01.lua

```diff
@@ -0,0 +1,223 @@
+end
+
+--------------------------------------------------------------------------
+-- Precondition:
+-- Check GetListOfPermissions response with empty externalConsentStatus array list.
+--------------------------------------------------------------------------
+Test["TEST_NAME_OFF".."_Precondition_GetListOfPermissions"] = function(self)
+  --hmi side: sending SDL.GetListOfPermissions request to SDL
+  local request_id = self.hmiConnection:SendRequest("SDL.GetListOfPermissions")
+  -- hmi side: expect SDL.GetListOfPermissions response
+  EXPECT_HMIRESPONSE(request_id,{
+      result = {
+        code = 0,
+        method = "SDL.GetListOfPermissions",
+        allowedFunctions = {
+          {name = "ConsentGroup001", allowed = nil},
+          {name = "ConsentGroup002", allowed = nil},
+          {name = "ConsentGroup003", allowed = nil},
+          {name = "ConsentGroup004", allowed = nil},
+          {name = "ConsentGroup005", allowed = nil}
+        },
+        externalConsentStatus = {}
+      }
+  })
+end
+
+--------------------------------------------------------------------------
+-- Precondition:
+-- HMI sends OnAppPermissionConsent
+--------------------------------------------------------------------------
+Test["TEST_NAME_OFF" .. "_Precondition_HMI_sends_OnAppPermissionConsent"] = function(self)
+  -- hmi side: sending SDL.OnAppPermissionConsent for applications
+  self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent", {
+      source = "GUI",
+      externalConsentStatus = {
+        {entityType = 1, entityID = 1, status = "OFF"},
+        {entityType = 6, entityID = 6, status = "OFF"},
+        {entityType = 7, entityID = 7, status = "OFF"},
+        {entityType = 9, entityID = 9, status = "OFF"}
+      }
+  })
+  self.mobileSession:ExpectNotification("OnPermissionsChange")
+  :ValidIf(function(_,data)
+      local validate_result_1 = common_functions_external_consent:ValidateHMIPermissions(data,
+        "SubscribeVehicleData", {allowed = {}, userDisallowed = {"BACKGROUND","FULL","LIMITED"}})
+      local validate_result_2 = common_functions_external_consent:ValidateHMIPermissions(data,
+        "SubscribeWayPoints", {allowed = {"BACKGROUND","FULL","LIMITED"}, userDisallowed = {}})
+      local validate_result_3 = common_functions_external_consent:ValidateHMIPermissions(data,
+        "SendLocation", {allowed = {"BACKGROUND","FULL","LIMITED"}, userDisallowed = {}})
+      return (validate_result_1 and validate_result_2 and validate_result_3)
+  end)
+end
+
+--------------------------------------------------------------------------
+-- Main check:
+-- RPC of Group001 is disallowed to process.
+--------------------------------------------------------------------------
+Test["TEST_NAME_OFF" .. "_MainCheck_RPC_of_Group001_is_disallowed"] = function(self)
+  local corid = self.mobileSession:SendRPC("SubscribeVehicleData",{rpm = true})
+  self.mobileSession:ExpectResponse(corid, {success = false, resultCode = "USER_DISALLOWED"})
```

Comment:
@vvvakulenko The result code is really USER_DISALLOWED and there is even defect that it is not properly sent